components/mediation-admin/org.wso2.carbon.message.processor/src/main/java/org/wso2/carbon/message/processor/service/MessageProcessorAdminService.java

```diff
@@ -755,4 +761,111 @@ public boolean validateAxis2ClientRepo(String input) throws AxisFault {
         }
     }
 
+    /**
+     * Gets the message from the associated queue.
+     *
+     * @param processorName message processor name.
+     * @return <code>message</code> returns message received from the queue as a string.
+     */
+    public String browseMessage(String processorName) throws AxisFault {
+        SynapseConfiguration configuration = getSynapseConfiguration();
+        MessageConsumer messageConsumer = getMessageConsumer(configuration,processorName);
+        String message = null;
+
+        try {
+            message = getMessageAsString(messageConsumer);
+        } catch (AxisFault e) {
+            handleException(log, "Failed to get message from the queue :", e);
+        }
+
+        messageConsumer.cleanup();
```

Comment:
Don't we need to have this in a finally block?

Reply:
Changed as suggested